src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py

```diff
@@ -0,0 +1,230 @@
+    blobName = match.group('blobName')
+    sasToken = match.group('sasToken')
+
+    if not accountName:
+        raise ValueError(
+            "Unable to parse the account name from the log URL: '{!s}'."
+            .format(logFileSas)
+        )
+
+    if not containerName:
+        raise ValueError(
+            "Unable to parse the container name from the log URL: '{!s}'."
+            .format(logFileSas)
+        )
+
+    if not blobName:
+        raise ValueError(
+            "Unable to parse the blob name from the log URL: '{!s}'."
+            .format(logFileSas)
+        )
+
+    if not sasToken:
+        raise ValueError(
+            "Unable to parse the SAS token from the log URL: '{!s}'."
+            .format(logFileSas)
+        )
+
+    blobService = AppendBlobService(account_name=accountName,
+                                    sas_token=sasToken)
+
+    byteSize = 1024*4
+    timeoutInMinutes = 10
+    timeoutInSeconds = timeoutInMinutes * 60
+    now = time.time()
+    stream_logs(byteSize, timeoutInSeconds, blobService,
+                containerName, blobName)
+
+
+def get_match(sasURL):
+    return re.search(
+        (r"http(s)?://(?P<accountName>.*?)\..*?/(?P<containerName>.*?)/"
+            r"(?P<blobName>.*?)\?(?P<sasToken>.*)"), sasURL)
+
+
+def stream_logs(byteSize, timeoutInSeconds, blobService,
+                containerName, blobName):
+
+    colorama.init()
+    stream = BytesIO()
+    metadata = dict()
+    start = 0
+    end = byteSize - 1
+    available = 0
+    lastModified = None
+
+    # Try to get the initial properties so there's no waiting.
+    # If the storage call fails, we'll just sleep and try again after.
+    try:
+        props = blobService.get_blob_properties(
+            container_name=containerName, blob_name=blobName)
+        metadata = props.metadata
+        available = props.properties.content_length
+        lastModified = props.properties.last_modified
+    except:
+        pass
+
+    while (blob_is_not_complete(metadata) or start < available):
+
+        while start < available:
+            try:
+                oldByteSize = len(stream.getvalue())
+                blobService.get_blob_to_stream(
+                    container_name=containerName,
+                    blob_name=blobName,
+                    start_range=start,
+                    end_range=end,
+                    stream=stream)
+
+                currBytes = stream.getvalue()
+                newByteSize = len(currBytes)
+                amountRead = newByteSize - oldByteSize
+                start += amountRead
+                end = start + byteSize - 1
+
+                # Only scan what's newly read. If nothing is read,
+                # default to 0.
+                minScanRange = max(newByteSize - amountRead - 1, 0)
+                for i in range(newByteSize - 1, minScanRange, -1):
+                    if currBytes[i-1:i+1] == b'\r\n':
+                        flush = currBytes[:i]  # won't print \n
```

Comment:
Curious, is `\n` always introduced? And why don't we want to print it?

Reply:
\r\n are synthetically inserted right now, it's just related to formatting.